state/azure/tablestorage/tablestorage.go

```diff
@@ -179,12 +179,10 @@ func (r *StateStore) Delete(req *state.DeleteRequest) error {
 	return err
 }
 
-func (r *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
+func (r *StateStore) Get(ctx context.Context, req *state.GetRequest) (*state.GetResponse, error) {
 	r.logger.Debugf("fetching %s", req.Key)
 	pk, rk := getPartitionAndRowKey(req.Key, r.cosmosDBMode)
-	getContext, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Here I see the default timeout time configured, named: `defaultxxxTimeout`, before because there is no context timeout mechanism. If `context.WithTimeout` method is not deleted here, then we need to introduce timeout parameter settings in the component specification. In this way, I can also compare the remaining time of the context passed by the request with the timeout set by the component, and choose a smaller one.

On the other hand, if you use `ctx, cancel:=context.WithTimeout(ctx, defaultXXXTimeout)`, it is not right. Because the request's upstream maybe already timeout and cancel the request, And I is invalid for downstream to continue processing the request.

Reply:
> On the other hand, if you use `ctx, cancel:=context.WithTimeout(ctx, defaultXXXTimeout)` it is not right. Because the request's upstream maybe already timeout and cancel the request, And I is invalid for downstream to continue processing the request.

Yes, that's what it should be. The request should end if:

- The timeout occurs
- Or if the upstream context is canceled: this is a valid reason for canceling the request because it means the receiver is not interested in that anymore

Reply:
Thanks. Fixed.